src/gtest/test_keystore.cpp

```diff
@@ -285,20 +289,22 @@ TEST(KeystoreTests, StoreAndRetrieveHDSeedInEncryptedStore) {
     TestCCryptoKeyStore keyStore;
     CKeyingMaterial vMasterKey(32, 0);
     GetRandBytes(vMasterKey.data(), 32);
-    HDSeed seedOut;
 
     // 1) Test adding a seed to an unencrypted key store, then encrypting it
-    auto seed = HDSeed::Random();
-    EXPECT_FALSE(keyStore.HaveHDSeed());
-    EXPECT_FALSE(keyStore.GetHDSeed(seedOut));
+    auto seed = MnemonicSeed::Random(BIP44_TESTNET_TYPE);
```

Comment:
The way that this is currently implemented, the legacy seed continues to be used as the source of addresses for when the user requests a Sapling-only address. But now that you ask this question, I'm trying to remember why we chose to do that, rather than use the `0x7FFFFFFE` account with the new mnemonic seed for generating Sapling-only addresses. Did we even discuss that as a possibility? If that account ID is reserved in zcashd as one for which UAs will *not* be produced, then it seems to me like we could switch the source of randomness over for Sapling as well and then I could remove more of the legacy seed support, which I think would be a good idea.